### PR TITLE
added settings to select statements

### PR DESCRIPTION
**Pull request by juansc**

### CODE REVIEW

File: select.go [hunk 1]
```diff
@@ -163,6 +164,8 @@ func (b *SelectStmt) Build(d Dialect, buf Buffer) error {
 			buf.WriteString(strconv.FormatInt(b.OffsetCount, 10))
 		}
 	}
+
+	b.settings.Build(d, buf)
```
Comment (khisakuni) [inline select.go:168]: Should this check if there are any settings before building it?

File: settings_test.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		expect   string
+	}{
+		{
+			name:     "test single setting",
+			settings: QuerySettings.Append(nil, "key", "value"),
+			expect:   "\nSETTINGS key = value",
+		},
+		{
+			name:     "test multiple setting",
+			settings: QuerySettings.Append(nil, "key1", "value1").Append("key2", "value2"),
```
Comment (khisakuni) [inline settings_test.go:24]: huh, didn't know you could do that!

File: select_test.go [hunk 3]
```diff
@@ -29,6 +29,30 @@ func TestSelectStmt(t *testing.T) {
 	require.Equal(t, 3, len(buf.Value()))
 }
 
+func TestSelectStmtWithSettings(t *testing.T) {
```
Comment (khisakuni) [inline select_test.go:32]: Can we add a test case for the case where we call two `Settings`? i.e. `Select("a").From("table").Settings("", "").Settings("", "")`

**Review by khisakuni — APPROVED** [review]

File: select.go (hunk 1 above)
Comment (juansc) [reply select.go:168]: We don't need to

**Pull request merged**